gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala

```diff
@@ -356,22 +356,20 @@ class TraversalSpec extends WordSpec with Matchers {
   }
 
   "union" should {
-
     "work for traversals with the same end type" in new Fixture {
-      val traversal = graph.V(4).union(
-        __[Vertex].in.value(Age),
-        __[Vertex].in.out.value(Age)
-      )
+      val traversal: GremlinScala[Int, _] =
+        graph.V(4).union(
+          _.in.value(Age),
+          _.in.out.value(Age)
+        )
 
```

Comment:
FWIW, I think the use of lambdas in lieu of anonymous traversals is brilliant. Took me a few minutes to grock what was going on. Did you consider creating a type alias for the function type, for readability sake? Something like:

```
type DetachedTraversal[A, B] = GremlinScala[A, HNil] ⇒ GremlinScala[B, _]
```
(I never quite know what to do with the `Labels` type parameter in these situations. In my own code I often cast it away when I no longer need the label.)

Also, the construct reminds me of the `Flow` type in  Akka streams.